casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -27,6 +27,12 @@ import scala.util.{Failure, Success, Try}
 
 //runtime is a SyncVar for thread-safety, as all checkpoints share the same "hot store"
 class RuntimeManager private (val emptyStateHash: ByteString, runtimeContainer: SyncVar[Runtime]) {
+  def setTimestamp(now: Long)(implicit scheduler: Scheduler): Unit = {
+    val runtime        = runtimeContainer.take()
```

Comment:
I see you ended up making the `setTimestamp ` a separate method. I still think it would be better to include timestamp as a parameter of `computeState` and `replayComputeState`.

Reply:
No I'm still WIP. I pushed this code specifically to show you the unsafeRunSync I want to remove